canary/webrtc-c/src/media-server-storage/Common.cpp

```diff
@@ -0,0 +1,1864 @@
+                DLOGE("Failed to cancel certificate pre-generation timer with: 0x%08x", retStatus);
+            }
+            pSampleConfiguration->pregenerateCertTimerId = MAX_UINT32;
+        }
+
+        timerQueueFree(&pSampleConfiguration->timerQueueHandle);
+    }
+
+    if (pSampleConfiguration->pPendingSignalingMessageForRemoteClient != NULL) {
+        // Iterate and free all the pending queues
+        stackQueueGetIterator(pSampleConfiguration->pPendingSignalingMessageForRemoteClient, &iterator);
+        while (IS_VALID_ITERATOR(iterator)) {
+            stackQueueIteratorGetItem(iterator, &data);
+            stackQueueIteratorNext(&iterator);
+            freeMessageQueue((PPendingMessageQueue) data);
+        }
+
+        stackQueueClear(pSampleConfiguration->pPendingSignalingMessageForRemoteClient, FALSE);
+        stackQueueFree(pSampleConfiguration->pPendingSignalingMessageForRemoteClient);
+        pSampleConfiguration->pPendingSignalingMessageForRemoteClient = NULL;
+    }
+
+    hashTableClear(pSampleConfiguration->pRtcPeerConnectionForRemoteClient);
+    hashTableFree(pSampleConfiguration->pRtcPeerConnectionForRemoteClient);
+
+    if (IS_VALID_MUTEX_VALUE(pSampleConfiguration->sampleConfigurationObjLock)) {
+        MUTEX_LOCK(pSampleConfiguration->sampleConfigurationObjLock);
+        locked = TRUE;
+    }
+
+    for (i = 0; i < pSampleConfiguration->streamingSessionCount; ++i) {
+        retStatus = gatherIceServerStats(pSampleConfiguration->sampleStreamingSessionList[i]);
+        if (STATUS_FAILED(retStatus)) {
+            DLOGW("Failed to ICE Server Stats for streaming session %d: %08x", i, retStatus);
+        }
+        freeSampleStreamingSession(&pSampleConfiguration->sampleStreamingSessionList[i]);
+    }
+    if (locked) {
+        MUTEX_UNLOCK(pSampleConfiguration->sampleConfigurationObjLock);
+    }
+    deinitKvsWebRtc();
+
+    SAFE_MEMFREE(pSampleConfiguration->pVideoFrameBuffer);
+    SAFE_MEMFREE(pSampleConfiguration->pAudioFrameBuffer);
+
+    if (IS_VALID_CVAR_VALUE(pSampleConfiguration->cvar) && IS_VALID_MUTEX_VALUE(pSampleConfiguration->sampleConfigurationObjLock)) {
+        CVAR_BROADCAST(pSampleConfiguration->cvar);
+        MUTEX_LOCK(pSampleConfiguration->sampleConfigurationObjLock);
+        MUTEX_UNLOCK(pSampleConfiguration->sampleConfigurationObjLock);
+    }
+
+    if (IS_VALID_MUTEX_VALUE(pSampleConfiguration->sampleConfigurationObjLock)) {
+        MUTEX_FREE(pSampleConfiguration->sampleConfigurationObjLock);
+    }
+
+    if (IS_VALID_MUTEX_VALUE(pSampleConfiguration->streamingSessionListReadLock)) {
+        MUTEX_FREE(pSampleConfiguration->streamingSessionListReadLock);
+    }
+
+    if (IS_VALID_MUTEX_VALUE(pSampleConfiguration->signalingSendMessageLock)) {
+        MUTEX_FREE(pSampleConfiguration->signalingSendMessageLock);
+    }
+
+    if (IS_VALID_CVAR_VALUE(pSampleConfiguration->cvar)) {
+        CVAR_FREE(pSampleConfiguration->cvar);
+    }
+
+#ifdef IOT_CORE_ENABLE_CREDENTIALS
```

Comment:
We are setting this up with env for our normal canaries, correct? We are not testing the IoT path for media server I assume?

Reply:
That's correct, this is just from the sample code.